react/index.d.ts

```diff
@@ -2610,9 +2610,7 @@ declare namespace React {
         isRequired: Validator<T>;
     }
 
-    interface ValidationMap<T> {
-        [key: string]: Validator<T>;
-    }
+    type ValidationMap<T> = { [K in keyof T]?: Validator<T> };
```

Comment:
Yes; the validator functions receive all of `T` along with a `string` parameter that specified which key that function is being asked to validate. Good question though!